### PR TITLE
Fixing assume roles for code deploy

### PR DESCRIPTION
**Pull request by Jetski5822**
Adding CodeDeployRoleName and fixing issue assuming roles, now we only assume a role is an ARN is passed through in RoleName

### CODE REVIEW

File: AWSPushAndDeploy/Program.cs [hunk 1]
```diff
@@ -10,15 +10,20 @@ class Program
     {
         static void Main(string[] args)
         {
+            Console.WriteLine("prog start opts:" + string.Join(",", args));
```
Comment (joshski) [inline AWSPushAndDeploy/Program.cs]: eww

File: TTC.Deployment.AmazonWebServices/Bundle.cs [hunk 2]
```diff
@@ -42,7 +46,10 @@ public string ApplicationSetName
 
         public string Bucket
         {
-            get { return _bucket; }
+            get {
+
+                Console.WriteLine("Bucket GET: " + _bucket);
```
Comment (joshski) [inline TTC.Deployment.AmazonWebServices/Bundle.cs]: eww

File: TTC.Deployment.AmazonWebServices/Deployer.cs [hunk 3]
```diff
@@ -50,14 +57,16 @@ public class Deployer
                                                     stsCredentials.SecretAccessKey,
                                                     stsCredentials.SessionToken);
 
-                credentials = sessionCredentials;
+                awsConfiguration.Credentials = sessionCredentials;
```
Comment (joshski) [inline TTC.Deployment.AmazonWebServices/Deployer.cs]: This is weird IMO. AWS configuration represents the command line arguments. Here, we update the object passed in to this method. I think this object should actually be immutable.

File: TTC.Deployment.AmazonWebServices/Bundle.cs [hunk 4]
```diff
@@ -165,19 +175,19 @@ public string Push(AmazonS3Client s3Client, AmazonCodeDeployClient codeDeployCli
...
 
             ZipFile.CreateFromDirectory(_bundleDirectory.FullName, tempPath, CompressionLevel.Optimal, false, Encoding.ASCII);
-
+            
             var allTheBuckets = s3Client.ListBuckets(new ListBucketsRequest()).Buckets;
 
             if (!allTheBuckets.Exists(b => b.BucketName == Bucket))
             {
-                s3Client.PutBucket(new PutBucketRequest { BucketName = Bucket, UseClientRegion = true });
+                s3Client.PutBucket(new PutBucketRequest { BucketName = Bucket, UseClientRegion = true, });
```
Comment (joshski) [inline TTC.Deployment.AmazonWebServices/Bundle.cs]: So much better with that trailing comma!

Comment (Jetski5822) [inline TTC.Deployment.AmazonWebServices/Bundle.cs]: good catch

File: TTC.Deployment.AmazonWebServices/Deployer.cs (hunk 3 above)
Comment (Jetski5822) [inline TTC.Deployment.AmazonWebServices/Deployer.cs]: Yeah I agree, I have changed this and cast internally

File: TTC.Deployment.AmazonWebServices/Bundle.cs (hunk 2 above)
Comment (Jetski5822) [inline TTC.Deployment.AmazonWebServices/Bundle.cs]: hehe removed

File: AWSPushAndDeploy/Program.cs (hunk 1 above)
Comment (Jetski5822) [inline AWSPushAndDeploy/Program.cs]: removed

**Pull request merged**